### PR TITLE
`SimulationBatch$addRunValues()` throws an error when `NaN`s are passed.

### LINKED ISSUE #1149: Update SimModel after Linux packages are created
Once the is issue is resolved

https://github.com/Open-Systems-Pharmacology/OSPSuite.CentOS/issues/13

### LINKED ISSUE #1150: `SimulationBatch$addRunValues` should throw an error when any value is `NaN`.
Otherwise we get:

With the simulation from here https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/1016

```
# Load the simulation from file - adjust the path accordingly
simulation <- loadSimulation("Vehicle.pkml")
# Get the paths of all molecules in the simulation
moleculePaths <- getAllMoleculePathsIn(container = simulation)
# Get the current intial values of all molecules
moleculesStartValues <- getQuantityValuesByPath(quantityPaths = moleculePaths,
                                                simulation = simulation)
# Create simulation batch and define all molecules as variable
simBatch <- createSimulationBatch(simulation = simulation,
                                  moleculesOrPaths = moleculePaths)
# Add run values
simBatch$addRunValues(initialValues = moleculesStartValues)
# Try to run
results <- runSimulationBatches(simBatch)
```

```
> results <- runSimulationBatches(simBatch)
Warning message:
In .getConcurrentSimulationRunnerResults(results = results, resultsIdSimulationIdMap = resultsIdSimulationBatchIdMap,  :
  Initial value of Vehicle|Organism|Liver|Pericentral|Interstitial|TMEFF2 was not set
```

### PR DESCRIPTION
**Pull request by PavelBal**
Fixes #1150

- `SimulationBatch$addRunValues()` will throw an error when any start value is `NaN`
- `SimulatioBatch` gets methods `getVariableParameters()` and `getVariableMolecules()` that return list of parameter resp. molecule paths that are defined variable.
- Added tests for `SimulationBatch`
- Styler

Fixes #1149

### CODE REVIEW

File: R/simulation-batch.R [hunk 1]
```diff
@@ -68,29 +69,62 @@ SimulationBatch <- R6::R6Class(
         stop(messages$errorOnlyOneValuesSetAllowed("parameterValues, initialValues"))
       }
 
+      # Check if any of the values is `NA`. If so, throw an error, as such
+      # values set will produce empty results
+      if (any(is.na(parameterValues))) {
```
Comment (PavelBal) [inline R/simulation-batch.R:74]: Here checking for NaNs

File: R/simulation-batch.R [hunk 2]
```diff
@@ -68,29 +69,62 @@ SimulationBatch <- R6::R6Class(
...
       return(batchRunValues$id)
     },
 
+    #' @description Returns a list of parameter paths that are variable in this batch.
+    #' @details The order of parameters is the same as the order of parameter
+    #' values added with `$addRunValues()` method.
+    #'
+    #' @return List of parameter paths, or `NULL` if no parameter is variable.
+    #' @export
+    getVariableParameters = function() {
```
Comment (PavelBal) [inline R/simulation-batch.R:94]: I think these functions are quite handy

File: R/simulation-batch.R [hunk 3]
```diff
@@ -68,29 +69,62 @@ SimulationBatch <- R6::R6Class(
...
+      simulationBatchOptions <- rClr::clrGet(self$ref, "SimulationBatchOptions")
```
Comment (PavelBal) [inline R/simulation-batch.R:95]: Could we cache `SimulationBatchOptions` as private field at `SimulationBatch` creation?

File: R/plot-observed-vs-simulated.R [hunk 4]
```diff
@@ -165,8 +165,12 @@ plotObservedVsSimulated <- function(dataCombined,
     plotConfiguration = obsVsPredPlotConfiguration
   )
 
-  return(plotObject + ggplot2::guides(shape = "none",
-                                      col = ggplot2::guide_legend(title = obsVsPredPlotConfiguration$legend$title$text,
-                                                                  title.theme = obsVsPredPlotConfiguration$legend$title$createPlotFont(),
-                                                                  override.aes = list(shape = overrideShapeAssignment$shapeAssn))))
+  return(plotObject + ggplot2::guides(
```
Comment (PavelBal) [inline R/plot-observed-vs-simulated.R:168]: Styler

File: R/plot-residuals-vs-simulated.R [hunk 5]
```diff
@@ -116,8 +116,12 @@ plotResidualsVsSimulated <- function(dataCombined,
       shape = "name"
     ),
     plotConfiguration = resVsPredPlotConfiguration
-  ) + ggplot2::guides(shape = "none",
-                      col = ggplot2::guide_legend(title = resVsPredPlotConfiguration$legend$title$text,
-                                                  title.theme = resVsPredPlotConfiguration$legend$title$createPlotFont(),
-                                                  override.aes = list(shape = overrideShapeAssignment$shapeAssn)))
+  ) + ggplot2::guides(
```
Comment (PavelBal) [inline R/plot-residuals-vs-simulated.R:119]: Styler

File: R/plot-residuals-vs-time.R [hunk 6]
```diff
@@ -115,8 +115,12 @@ plotResidualsVsTime <- function(dataCombined,
       shape = "name"
     ),
     plotConfiguration = resVsTimePlotConfiguration
-  ) + ggplot2::guides(shape = "none",
-                    col = ggplot2::guide_legend(title = resVsTimePlotConfiguration$legend$title$text,
-                                                title.theme = resVsTimePlotConfiguration$legend$title$createPlotFont(),
-                                                override.aes = list(shape = overrideShapeAssignment$shapeAssn)))
+  ) + ggplot2::guides(
```
Comment (PavelBal) [inline R/plot-residuals-vs-time.R:118]: Styler

File: R/utilities-data-combined.R [hunk 7]
```diff
@@ -209,8 +209,10 @@ calculateResiduals <- function(dataCombined,
   # Add predicted values
   # the approx function with a default rule = 1 argument returns NA for extrapolated points
   pairedData <- dplyr::mutate(pairedData,
-    "yValuesSimulated" = approx(simulatedData$xValues, simulatedData$yValues,
-                                observedData$xValues)$y
+    "yValuesSimulated" = approx(
```
Comment (PavelBal) [inline R/utilities-data-combined.R:212]: Styler

File: tests/testthat/test-simulation-batch.R [hunk 8]
```diff
@@ -0,0 +1,110 @@
+tolerance <- 0.0001
+sim <- loadTestSimulation("simple", loadFromCache = TRUE)
+
+test_that("It returns correct lists for variable parameters or molecules, or NULL
```
Comment (PavelBal) [inline tests/testthat/test-simulation-batch.R:4]: New test for `getVariableParameters()` and `getVariableMolecules()` methods

File: tests/testthat/test-simulation-batch.R [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+  simBatch <- createSimulationBatch(
+    simulation = sim,
+    moleculesOrPaths = moleculePaths
+  )
+
+  expect_null(simBatch$getVariableParameters())
+  expect_equal(simBatch$getVariableMolecules(), moleculePaths)
+})
+
+test_that("It throws an error when any of the added molecule start values is NaN", {
```
Comment (PavelBal) [inline tests/testthat/test-simulation-batch.R:30]: Test for throwing an error when `NaN` is passed as start value.

File: tests/testthat/test-simulation-batch.R [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+    simulation = sim,
+    parametersOrPaths = parameterPaths
+  )
+  # Add run values
+  expect_error(simBatch$addRunValues(parameterValues = parameterValues),
+    regexp = messages$simBatchStartValueNaN(parameterPaths[[1]])
+  )
+})
+
+test_that("It actually sets new initial values", {
```
Comment (PavelBal) [inline tests/testthat/test-simulation-batch.R:72]: We never actually testet this, lol

**Comment by PavelBal** [comment]
@msevestre Could you review?

**Comment by Yuri05** [comment]
@PavelBal Can you fix https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/1149 as well?
Just increment the SimModel version in **packages.centOS7.config**, **packages.ubuntu18.config** and **packages.win.config** to **4.0.0.53**

**Comment by PavelBal** [comment]
@Yuri05 Done

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**